eng/Versions.props

```diff
@@ -114,7 +114,7 @@
           so target one that matches the version in minimumMSBuildVersion.
 
           This avoids the need to juggle references to packages that have been updated in newer MSBuild. -->
-    <MicrosoftBuildPackageVersion Condition="exists('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion') and '$(DotNetBuildFromSource)' != 'true'">$([System.IO.File]::ReadAllText('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion').Trim())</MicrosoftBuildPackageVersion>
+    <MicrosoftBuildPackageVersion Condition="exists('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion') and '$(DotNetBuildFromSource)' != 'true'">17.4.6</MicrosoftBuildPackageVersion>
```

Comment:
nit:
```suggestion
    <MicrosoftBuildPackageVersion Condition="'$(DotNetBuildFromSource)' != 'true'">17.4.6</MicrosoftBuildPackageVersion>
```

Reply:
Yeah, wasn't sure whether to remove or not since it's no longer relevant. I went for the smaller change. BTW, I have to do 17.4.8 it looks like as I realized there is another binary impacted so I'll have an update to this PR coming soon.